### ISSUE TITLE
Implement .catch() in ParsePromise

### ISSUE THREAD

**Issue by AriaFallah** [ParsePlatform/Parse-SDK-JS#183]
This is standard in every other promise implementation I've seen except for Parse. I have to do this to be able to use parse promises with any library that expects a normal promise:

```javascript
new Promise((resolve, reject) => promise.then((data) => resolve(data), (err) => reject(err)))
```

**Comment by andrewimm** [comment]
See `Parse.Promise.fail()` for now. Keep in mind that the official Promises/A+ spec only governs how `.then()` operates, it does not dictate how promises should be constructed, or any additional functionality that should be added.

We recently enabled ES6-style promise creation for 1.7.0, and at the time considered making `.catch` an alias for `.fail`, but didn't want to commit to adding extraneous APIs at the time. If we create an implementation for `.catch`, we'd also probably want to implement `Promise.all` (which behaves slightly differently from `Parse.Promise.wait`) and `Promise.race`. Eventually, we'd like to have our Promise implementation completely interchangeable with the ES6 version, but these are changes that probably won't occur until at least 1.8.0 (because they represent new APIs). You can see #57 for more info on this.

**Comment by dwilt** [comment]
+1 on this and the interchangeability of the ES6 syntax and Parse. 

@AriaFallah I was having trouble getting the `.catch` to work as well and couldn't figure out why then realized they worked with the native promises I made but not with the Parse ones. To fix, I've been just wrapping all of my Parse calls with the new Promise like you've mentioned:

```javascript
return new Promise((resolve, reject) => {
    Parse.Cloud.run('getUser', {
        id
    }).then(resolve, reject);
});
```

**Comment by andrewimm** [comment]
It'll probably still wait until 1.8.0, but I went ahead and implemented the missing APIs in #187
